File: LLD/InterviewExamples/Blinkit/Java/src/repositories/OrderRepository.java
package repositories;

import models.Order;

import java.util.ArrayList;
import java.util.List;

public class OrderRepository {

    private final List<Order> orderList;

    public OrderRepository() {
        orderList = new ArrayList<>();
    }

    public List<Order> getOrderList() {
        return this.orderList;
    }

    public Order getOrderById(String id) {
        for(Order order : orderList) {
            if(order.getId().equals(id)) return order;
        }

        return null;
    }
}

File: LLD/InterviewExamples/Blinkit/Node/src/repositories/orderRepository.js
const Order = require('../models/order')

const orderList = []

module.exports = {
    getOrderList : () => {
        return orderList;
    },

    addOrder : (order) => {
        orderList.add(order);
    },

    getOrderById : (id) => {
        return orderList.filter((order) => order.getId().equals(id))[0];
    }
}